lib/url.js

```diff
@@ -948,13 +953,23 @@ Url.prototype.resolveObject = function resolveObject(relative) {
 
 Url.prototype.parseHost = function parseHost() {
   var host = this.host;
-  var port = portPattern.exec(host);
-  if (port) {
-    port = port[0];
-    if (port !== ':') {
-      this.port = port.slice(1);
+  var port;
+  // Adding support for the unix socket hostname and port resolution
+  if (!host.startsWith('unix:')) {
+    port = portPattern.exec(host);
+    if (port) {
+      port = port[0];
+      if (port !== ':') {
+        this.port = port.slice(1);
+      }
+      host = host.slice(0, host.length - port.length);
+    }
+  } else {
+    var hostSplit = host.split(':');
+    if (hostSplit[1]) {
+      this.port = hostSplit[1] + (host.endsWith(':') ? ':' : '');
     }
-    host = host.slice(0, host.length - port.length);
+    host = hostSplit[0];
```

Comment:
If we were to support parsing http over unix socket, I'd prefer adding a new field `socketPath` instead of replacing values at the `host` and `port`, which make sense for TCP only.